File: Python/Python OOP/Inheritance.py
# python - Inheritance

class A:
    def __init__(self) -> None:
        print('class A constructor')

# inherit class A
class B (A):
    def __init__(self) -> None:
        print('class B constructor')

        # parent class super methode
        super().__init__()

class D:
    def __init__(self) -> None:
        pass

# multiple inheritance
class C (A,D):
    def __init__(self) -> None:
        super().__init__()

b = B()

File: TypeScript/TS OOPs/Inheritance.ts
// TS - Inheritance 

class A {
    public a: number = 10
    private p: number = 93
    protected b: number = 22
}

class B extends A {
    constructor() {
        super()
    }

    display() {
        console.log(`${this.a} ${this.b} `)
    }
}

